src/entities/pool.ts

```diff
@@ -772,99 +774,149 @@ export class Pool {
+    dapp: Dapp,
     tokenId: string,
     amountA: BigNumber | string,
     amountB: BigNumber | string,
     options: any = null
   ): Promise<any> {
-    const iNonfungiblePositionManager = new ethers.utils.Interface(
-      INonfungiblePositionManager.abi
-    );
-    const increaseLiquidityTxData = iNonfungiblePositionManager.encodeFunctionData(
-      Transaction.INCREASE_LIQUIDITY,
-      [[tokenId, amountA, amountB, 0, 0, deadline]]
-    );
+    let txData;
+    let dappAddress;
+    if (dapp === Dapp.UNISWAPV3) {
+      dappAddress = nonfungiblePositionManagerAddress[this.network];
+      const abi = new ethers.utils.Interface(INonfungiblePositionManager.abi);
+      txData = abi.encodeFunctionData(Transaction.INCREASE_LIQUIDITY, [
+        [tokenId, amountA, amountB, 0, 0, deadline]
+      ]);
+    } else if (dapp === Dapp.ARRAKIS) {
+      dappAddress = routerAddress[this.network][dapp];
+      const abi = new ethers.utils.Interface(IArrakisV1RouterStaking.abi);
+      txData = abi.encodeFunctionData(Transaction.ADD_LIQUIDITY_STAKE, [
+        tokenId,
+        amountA,
+        amountB,
+        0,
+        0,
+        this.address
+      ]);
+    } else {
+      throw new Error("dapp not supported");
+    }
+
     const tx = await this.poolLogic.execTransaction(
-      nonfungiblePositionManagerAddress[this.network],
-      increaseLiquidityTxData,
+      dappAddress,
+      txData,
       options
     );
     return tx;
   }
 
   /**
-   * Claim fees of an UniswapV3 liquidity pool
-   * @param {string} tokenId Token Id of UniswapV3 position
+   * Claim fees of an UniswapV3 liquidity or Arrakis pool
+   * @param {Dapp} dapp Platform either UniswapV3 or Arrakis
+   * @param {string} tokenId Token Id of UniswapV3 or Arrakis position
    * @param {any} options Transaction options
    * @returns {Promise<any>} Transaction
    */
-  async claimFeesUniswapV3(tokenId: string, options: any = null): Promise<any> {
-    const iNonfungiblePositionManager = new ethers.utils.Interface(
-      INonfungiblePositionManager.abi
-    );
-    const collectTxData = iNonfungiblePositionManager.encodeFunctionData(
-      Transaction.COLLECT,
-      [[tokenId, this.address, MaxUint128, MaxUint128]]
-    );
+  async claimFees(
+    dapp: Dapp,
+    tokenId: string,
+    options: any = null
+  ): Promise<any> {
+    let txData;
+    let contractAddress;
+    if (dapp === Dapp.UNISWAPV3) {
+      contractAddress = nonfungiblePositionManagerAddress[this.network];
+      const iNonfungiblePositionManager = new ethers.utils.Interface(
+        INonfungiblePositionManager.abi
+      );
+      txData = iNonfungiblePositionManager.encodeFunctionData(
+        Transaction.COLLECT,
+        [[tokenId, this.address, MaxUint128, MaxUint128]]
+      );
+    } else if (dapp === Dapp.ARRAKIS) {
+      contractAddress = tokenId;
+      const abi = new ethers.utils.Interface(ILiquidityGaugeV4.abi);
+      txData = abi.encodeFunctionData("claim_rewards()", []);
```

Comment:
is it put `()` by mistake?

Reply:
No mistake, you can overload functions in Solidity, so the encoder knows to use the implementation without input parameter when you use ()
https://www.tutorialspoint.com/solidity/solidity_function_overloading.htm

Reply:
make sense then